helix-view/src/editor.rs

```diff
@@ -440,7 +440,10 @@ impl Editor {
     /// Refreshes the language server for a given document
     pub fn refresh_language_server(&mut self, doc_id: DocumentId) -> Option<()> {
         let doc = self.documents.get_mut(&doc_id)?;
-        doc.detect_language(self.syn_loader.clone());
+        // try detection only if language is not already set for this document.
+        if doc.language.is_none() {
+            doc.detect_language(self.syn_loader.clone());
+        }
```

Comment:
Let's move this out of the method, the only use is in https://github.com/helix-editor/helix/blob/f2dd3d446914ec0953cd7e056d430fc3b959e75d/helix-term/src/commands/typed.rs#L200-L203= where the point is that we detect the filetype if a file was a scratch buffer before (i.e. make a new file, write some code, then save with `.rs` extension and it should get highlighted). I would just call `detect_language` there.

Reply:
done.